readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -395,6 +397,57 @@ def build(self):
             raise BuildEnvironmentError('No TeX files were found')
 
         # Run LaTeX -> PDF conversions
+        if self.project.has_feature(Feature.USE_PDF_LATEXMK):
+            return self._build_latexmk(cwd, latex_cwd)
+
+        return self._build_pdflatex(tex_files, latex_cwd)
+
+    def _build_latexmk(self, cwd, latex_cwd):
+        # These steps are copied from the Makefile generated by Sphinx >= 1.6
+        # https://github.com/sphinx-doc/sphinx/blob/master/sphinx/texinputs/Makefile_t
+        latex_path = Path(latex_cwd)
+        images = []
+        for extension in ('png', 'gif', 'jpg', 'jpeg'):
+            images.extend(latex_path.glob(f'*.{extension}'))
+
+        # FIXME: instead of checking by language here, what we want to check if
+        # ``latex_engine`` is ``platex``
+        pdfs = []
+        if self.project.language == 'ja':
+            pdfs = latex_path.glob('*.pdf')
+
+        for image in itertools.chain(images, pdfs):
+            self.run(
+                'extractbb',
+                image.name,
+                cwd=latex_cwd,
+                record=False,
+            )
+
+        rcfile = 'latexmkrc'
+        if self.project.language == 'ja':
+            rcfile = 'latexmkjarc'
+
+        cmd_ret = self.run(
+            'latexmk',
+            '-r',
+            rcfile,
+
+            # FIXME: check for platex here as well
```

Comment:
I'm confused about platex and latexmk

Reply:
`latexmk` uses internally `pdflatex` or `xelatex` or `lualatex` or `platex` (defined by [`latex_engine`](http://www.sphinx-doc.org/en/master/usage/configuration.html#confval-latex_engine)